File: imagegame_ui/src/services/BadgeService.js
import http from "../http-common"; 

class BadgeService {
  getAllBadges(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/badge/badges`, searchDTO);
  }

  get(badgeId) {
    return this.getRequest(`/badge/${badgeId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/badge?field=${matchData}`, null);
  }

  addBadge(data) {
    return http.post("/badge/addBadge", data);
  }

  update(data) {
  	return http.post("/badge/updateBadge", data);
  }
  
  uploadImage(data,badgeId) {
  	return http.postForm("/badge/uploadImage/"+badgeId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new BadgeService();

File: src/main/java/com/imagegame/service/BadgeService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.Badge;
import com.imagegame.dto.BadgeDTO;
import com.imagegame.dto.BadgeSearchDTO;
import com.imagegame.dto.BadgePageDTO;
import com.imagegame.dto.BadgeConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface BadgeService extends GenericService<Badge, Integer> {

	List<Badge> findAll();

	ResultDTO addBadge(BadgeDTO badgeDTO, RequestDTO requestDTO);

	ResultDTO updateBadge(BadgeDTO badgeDTO, RequestDTO requestDTO);

    Page<Badge> getAllBadges(Pageable pageable);

    Page<Badge> getAllBadges(Specification<Badge> spec, Pageable pageable);

	ResponseEntity<BadgePageDTO> getBadges(BadgeSearchDTO badgeSearchDTO);
	
	List<BadgeDTO> convertBadgesToBadgeDTOs(List<Badge> badges, BadgeConvertCriteriaDTO convertCriteria);

	BadgeDTO getBadgeDTOById(Integer badgeId);







}





